patches/server/1059-Implement-configurable-per-world-void-damage-thresho.patch

```diff
@@ -0,0 +1,103 @@
+         public boolean generateFlatBedrock = false;
+         public FrostedIce frostedIce;
++        public VoidDamage voidDamage;
++
++        public class VoidDamage extends ConfigurationPart {
++            public boolean enabled = true;
++            public Damage damage;
++
++            public class Damage extends ConfigurationPart {
++                public double heightThreshold = -128.0;
++                public float amount = 4.0F;
++            }
++        }
+ 
+         public class FrostedIce extends ConfigurationPart {
+             public boolean enabled = true;
+diff --git a/src/main/java/net/minecraft/world/entity/Entity.java b/src/main/java/net/minecraft/world/entity/Entity.java
+index bc45c74987974b4828201e06fc8b1f3fbc0af8b4..305b1d80559077333cc5a31a7b217cf2f764cf5d 100644
+--- a/src/main/java/net/minecraft/world/entity/Entity.java
++++ b/src/main/java/net/minecraft/world/entity/Entity.java
+@@ -310,7 +310,7 @@ public abstract class Entity implements SyncedDataHolder, Nameable, EntityAccess
+     protected int boardingCooldown;
+     @Nullable
+     private Entity vehicle;
+-    private Level level;
++    Level level;
+     public double xo;
+     public double yo;
+     public double zo;
+@@ -931,8 +931,12 @@ public abstract class Entity implements SyncedDataHolder, Nameable, EntityAccess
+     }
+ 
+     public void checkBelowWorld() {
++        if (!this.level.paperConfig().environment.voidDamage.enabled) {
++            return;
++        }
++
+         // Paper start - Configurable nether ceiling damage
+-        if (this.getY() < (double) (this.level.getMinBuildHeight() - 64) || (this.level.getWorld().getEnvironment() == org.bukkit.World.Environment.NETHER
++        if (this.getY() < this.level.paperConfig().environment.voidDamage.damage.heightThreshold  || (this.level.getWorld().getEnvironment() == org.bukkit.World.Environment.NETHER
+             && this.level.paperConfig().environment.netherCeilingVoidDamageHeight.test(v -> this.getY() >= v)
+             && (!(this instanceof Player player) || !player.getAbilities().invulnerable))) {
+             // Paper end - Configurable nether ceiling damage
+diff --git a/src/main/java/org/bukkit/craftbukkit/CraftWorld.java b/src/main/java/org/bukkit/craftbukkit/CraftWorld.java
+index cd3381d8225e0322b3e285b3c4aa0a1265a91cc0..64b4ee8bb1adf735efda00dfc87f71bc2c60f8bb 100644
+--- a/src/main/java/org/bukkit/craftbukkit/CraftWorld.java
++++ b/src/main/java/org/bukkit/craftbukkit/CraftWorld.java
+@@ -163,8 +163,30 @@ public class CraftWorld extends CraftRegionAccessor implements World {
+     private final Object2IntOpenHashMap<SpawnCategory> spawnCategoryLimit = new Object2IntOpenHashMap<>();
+     private final CraftPersistentDataContainer persistentDataContainer = new CraftPersistentDataContainer(CraftWorld.DATA_TYPE_REGISTRY);
+     private net.kyori.adventure.pointer.Pointers adventure$pointers; // Paper - implement pointers
++    private float voidDamageAmount;
```

Comment:
These values seem completely unused, the logic in entity uses the world config, no the CB worlds values.